internal/inc/file_internal.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef FILE_INTERNAL_H
+#define FILE_INTERNAL_H
+
+#ifdef __cplusplus
+#include <cstdint>
+#else
+#include <stdint.h>
+#endif
+
+#include "umock_c/umock_c_prod.h"
+#include "azure_macro_utils/macro_utils.h"
+#include "constbuffer.h"
+#include "file.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct FILE_WRITE_DATA_TAG
+{
+    FILE_WRITE_CB user_callback;
+    void* user_context;
+    CONSTBUFFER_HANDLE source;
+}FILE_WRITE_DATA;
+
+typedef struct FILE_READ_DATA_TAG
+{
+    FILE_READ_CB user_callback;
+    void* user_context;
+    uint32_t size;
+    unsigned char* destination;
+}FILE_READ_DATA;
+
+typedef union FILE_IO_DATA_TAG

```

Comment:
>union [](start = 8, length = 5)

why is this union needed?

Reply:
Because we don't know what sort of data is expected on the windows callback. Since I separated the callbacks for linux, I'm moving this union to file_win32.c.

Also bsdl_file_ll does it this way.

---
In reply to: [450466771](https://github.com/Azure/azure-c-pal/pull/7#discussion_r450466771) [](ancestors = 450466771)